examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,294 @@
+# %%
+"""
+Crane moving a load
+===================
+
+A load is moved by a crane. The load is rotably connected to the crane by a
+massless rod. The goal is to move the load from the initial position to the
+final position in the shortest possible time. The load must not 'overswing'
+the final position. The load is moved by a force applied to its
+suspension point. The jib of the crane is extended in the horizontal
+X direction.
+
+**Constants**
+
+- l: length of the rod attaching the load to the crane [m]
+- m1: mass of mover attached to the arm of the crane [kg]
+- m2: mass of the load [kg]
+- g: acceleration due to gravity [m/s²]
+
+**States**
+
+- xc: x-coordinate of the mover [m]
+- xl: x-coordinate of the load [m]  dependent coordinate
+- yl: y-coordinate of the load [m]  dependent coordinate
+- q: angle of the rod [rad]
+- uxc: velocity of the mover in x-direction [m/s]
+- uxl: velocity of the load in x-direction [m/s] dependent speed
+- uyl: velocity of the load in y-direction [m/s] dependent speed
+- u: angular velocity of the rod [rad/s]
+
+**Specifieds**
+
+- F: force applied to the mover [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from scipy.interpolate import CubicSpline
+from opty.direct_collocation import Problem
+from opty.utils import parse_free
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+from matplotlib import patches
+
+# %%
+# Set up Kane's equations of motion.
+# Just the standard way of doing it.
+
+N, A = sm.symbols('N A', cls = me.ReferenceFrame)
+t = me.dynamicsymbols._t
+O, P1, P2 = sm.symbols('O P1 P2', cls = me.Point)
+O.set_vel(N, 0)
+xc, xl, yl, q, uxc, uxl, uyl, u, F = me.dynamicsymbols('xc, xl, yl, q, uxc,' +
+    'uxl, uyl, uq, F')
+l, m1, m2, g = sm.symbols('l, m1, m2, g')
+
+A.orient_axis(N, q, N.z)
+A.set_ang_vel(N, u * N.z)
+
+P1.set_pos(O, xc * N.x)
+P1.set_vel(N, uxc * N.x)
+
+P2.set_pos(P1, -l * A.y)
+P2.v2pt_theory(P1, N, A)
+
+P1a = me.Particle('P1a', P1, m1)
+P2a = me.Particle('P2a', P2, m2)
+
+BODY = [P1a, P2a]
```

Comment:
```suggestion
bodies = [P1a, P2a]
```

Use PEP8 naming conventions if at all possible.